content/docs/userspace/http-api-guide.md

```diff
@@ -0,0 +1,649 @@
+which can then be used for communications with the ship.
+
+**Note:** An Urbit ship will deny CORS requests from external URLs by default.
+In order to allow CORS for an external web app, the user will need to run the
+following in the dojo (replacing `https://example.com` with the address in
+question):
+
+```
+|cors-approve 'https://example.com'
+```
+
+#### Example
+
+Here's a simple example that will run `authenticate` for a fakezod when the
+_Connect_ button is pressed:
+
+```
+<html>
+  <head>
+    <script src="https://unpkg.com/@urbit/http-api"></script>
+  </head>
+  <body>
+    <button id="start" type="button" onClick="connect()" >Connect</button>
+  </body>
+  <script>
+    async function connect() {
+      window.api = await UrbitHttpApi.Urbit.authenticate({
+          ship: "zod",
+          url: "localhost:8080",
+          code: "lidlut-tabwed-pillex-ridrup",
+          verbose: true
+      });
+      document.body.innerHTML = "Connected!";
+    };
+  </script>
+</html>
+```
+
+`window.api` can then be used for further communications.
+
+## Functions
+
+The various functions you can use in the `Urbit` object are described below.
+
+### Poke
+
+For poking a ship, the `Urbit` class in `http-api` includes a `poke` function.
+The `poke` function takes six arguments in a object:
+
+| Argument    | Type        | Description                                                                                                                  | Example                 |
+| ----------- | ----------- | ---------------------------------------------------------------------------------------------------------------------------- | ----------------------- |
+| `app`       | `string`    | The Gall agent to poke.                                                                                                      | `"graph-store"`         |
+| `mark`      | `string`    | The mark of the data to poke the agent with.                                                                                 | `"graph-update-3"`      |
+| `json`      | any JSON    | The data to poke the agent with.                                                                                             | `{foo: "bar", baz: 42}` |
+| `ship`      | `string`    | The Urbit ID (`@p`) of the ship without the `~`. This may be ommitted if it's already been set for the whole `Urbit` object. | `"sampel-palnet"`       |
+| `onSuccess` | A function. | This is called if the poke succeeded (the ship ack'd the poke).                                                              | `someFunction`          |
+| `onError`   | A function. | This is called if the poke failed (the ship nack'd the poke).                                                                | `anotherFunction`       |
+
+#### Example
+
+This example lets you "hi" your ship with a given message, which will be printed
+in the terminal. It pokes the `hood` agent (the system agent) with a `helm-hi`
```

Comment:
Perhaps clarify that `|hi` is a %hood operation a bit more?  I.e. that this isn't general terminal command execution.